pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/PendingAckHandleImpl.java

```diff
@@ -945,12 +955,35 @@ public void completeHandleFuture() {
     }
 
     public void exceptionHandleFuture(Throwable t) {
+        if(isRetryableException(t)) {
+            long retryTime = backoff.next();
+            log.warn("[{}][{}] Failed to init transaction pending ack handler. It will be retried in {} Ms",
+                    persistentSubscription.getTopic().getName(), subName, retryTime, t);
+            transactionOpTimer.newTimeout((timeout) -> init(), retryTime, TimeUnit.MILLISECONDS);
+            return;
+        }
+        log.error("PendingAckHandleImpl init fail! TopicName : {}, SubName: {}", topicName, subName, t);
+        handleCacheRequest();
+        changeToErrorState();
+        this.pendingAckStoreFuture.completeExceptionally(new TransactionPendingAckException
+                .TransactionPendingAckStoreInitException(String.format("PendingAckHandleImpl init fail! TopicName : "
```

Comment:
We already defined the `TransactionPendingAckStoreInitException`. The message of the exception should be provided by the `TransactionPendingAckStoreInitException`. And I don't think we need to add the topic name, or subscription name here. The caller knows which topic and subscription they are requesting.

Reply:
First, the definition of `TransactionPendingAckStoreInitException` does not provide the error message.
Secondly, this might not be an end-to-end request. It could happen at topic loading.
```
    public static class TransactionPendingAckStoreInitException extends TransactionPendingAckException {
        public TransactionPendingAckStoreInitException(String message) {
            super(message);
        }

        public TransactionPendingAckStoreInitException(String message, Throwable throwable) {
            super(message, throwable);}
    }
```

Reply:
Sorry, I mean you have created a specific exception "TransactionPendingAckStoreInitException", it should provide a specific error message. Because you will not have an error message like "Transaction pending ack close failed".

Reply:
Thanks for your explanation. I understand what you mean.